kotlin/internal/opts.bzl

```diff
@@ -86,6 +86,13 @@ _KOPTS = {
         ),
         type = attr.bool,
     ),
+    "opt_in": struct(
```

Comment:
Should this be in _KOPTS? It also felt weird to add it to a structure that is essentially identical but I suppose the important distinction is that toolchain overrides do not apply to this (based on compile.bzl)

Reply:
This is sort of the tricky bit -- we only have one options rule for the moment. I'm inclined to split it out into two rules (lib and toolchain). What are your thoughts? 

What I'd like to avoid is the toolchain having optins that are ignored or error -- the idea is to make the distinction clear.

I'm also willing to just let optins be on the toolchain.

What are your thoughts?

Reply:
@restingbull i think two distinct ones sounds fine to me - would we duplicate the flags in each that can be overridden or would is there someway to embed all toolchain flags into rule flags (if thats the desired behavior -not aware of toolchain flags that shouldnt be overriden by rules)?

Reply:
@restingbull just wanted to bump this as im not sure if you saw or not - its been a bit. Any thoughts on duplicating all flags vs embedding toolchain to be a strict superset or rule flags? The overlap is enticing to mitigate but may not be worthwhile